jmclient/jmclient/fidelity_bond.py

```diff
@@ -114,9 +113,9 @@ def _verify_signature(message, signature, pubkey):
 
     @classmethod
     def parse_and_verify_proof_msg(cls, maker_nick, taker_nick, data):
-        if not fidelity_bond_sanity_check.fidelity_bond_sanity_check(data):
-            raise ValueError("sanity check failed")
         decoded_data = base64.b64decode(data)
```

Comment:
Two things:
1/ We need to catch a decode failure Exception here.

2/ If you look at `fidelity_bond_sanity_check`, the call to `decode` includes a `validate=True` argument (basically a stricter check of validity).

Reply:
> 1/ We need to catch a decode failure Exception here.

You mean catching `binascii.Error` and raising `ValueError` instead? (will happen if `data` is incorrectly padded)

> 2/ If you look at `fidelity_bond_sanity_check`, the call to `decode` includes a `validate=True` argument (basically a stricter check of validity).

Will add `validate=True`.

Reply:
> You mean catching binascii.Error and raising ValueError instead? (will happen if data is incorrectly padded)

I wasn't thinking very clearly but I guess I do mean exactly that, or more generally, just catch any Exception and raise ValueError; my only reason for saying it, is to preserve the behaviour we had before.

Reply:
Not sure about catching any exceptions. If it raises something else, we have no idea what happened, is it kinda `ValueError`, so probably it's bettter to just die.

Reply:
There's two general ways of looking at "unexpected exceptions" and they're kind of opposite :)  : 

1/ Better not swallow unexpected exceptions, because it means something is very wrong in the code, so it's dangerous. better panic and quit.

2/ Swallow *any* exception to avoid crashes.

While 1/ is usually the right way, in cases where your code receives input from untrusted external sources (as is the case in some parts of Joinmarket - messages from counterparties), 2/ applies. I think that is the case here, because we are reading serialized fidelity bond messages from untrusted counterparties.

So that's probably why the existing code caught `Exception` and returned False, and we turned it into `ValueError`. Probably best to just stick with that?

Reply:
I disagree here. According to [docs](https://docs.python.org/3/library/base64.html#base64.b64decode), only `binascii.Error` can be raised because of invalid `data`. If something else is raised, it means API has changed and our code is not aware of that.

Reply:
Oh yes, that's a good point. Then I agree that `binascii.Error` is better.